### ISSUE TITLE
enable "promisify" / "denodify" of function that loads the JSAPI

### ISSUE THREAD

**Issue by tomwayson** [Esri/esri-loader#30]
As a stopgap to #8 we could make it easier for consumers and the wrapper libraries to [promisify](http://bluebirdjs.com/docs/api/promise.promisify.html) / [denodify](https://emberjs.com/api/ember/2.14/classes/RSVP/methods/denodeify?anchor=denodeify) the bootstrap function by switching the order of the arguments so that the callback comes last.

Instead of making a breaking change for this, I propose deprecating `bootstrap()` in favor of a new function w/ the correct argument order. I've never been a huge fan of the name "bootstrap" anyway. I'm leaning toward `loadScript()` but am open to suggestion.

I also propose that we make the changes suggested in #28 in this new function since those might be considered breaking. As long as we're making what would otherwise be breaking changes in a new function called `loadScript()`, I suggest that callback for this new function always return the HTMLScriptElement instead of `window.require` in order to address the issue I brought up in https://github.com/Esri/esri-loader/issues/28#issuecomment-326130982. Or maybe it would be more useful to return the [verson](https://developers.arcgis.com/javascript/latest/api-reference/esri-kernel.html#version) of the JSAPI that was loaded?

My goal for this is to reduce [these ~50 lines in ember-esri-loader](https://github.com/Esri/ember-esri-loader/blob/7dfaccc6717b4ee22dab768009f5ea751902191f/addon/services/esri-loader.js#L33-L85) to:

```js
const loadScript = RSVP.denodeify(esriLoader.loadScript);
this._loadPromise = loadScript(options)
.then(version => {
  // update the isLoaded computed property
  this.notifyPropertyChange('isLoaded');
  // TODO: remove success: true next major version bump?
  return { success: true, version };
});
```

**Comment by tomwayson** [comment]
While we're at it, let's take an optional `url` instead of `options`, so:

```
// default, latest JSAPI 4.x
esriLoader.loadScript(callback);
// or use a specific version/location of JSAPI
esriLoader.loadScript('https://js.arcgis.com/3.21', callback);
```

Originally I used options b/c I thought we might want to things like letting users set other script attributes like `defer` on the script tag, but no one seems to be asking for that. If later on we decide that we need options we can use [TS union types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#union-types) to allow passing in a string (url) _or_ object (options).

**Comment by tomwayson** [comment]
Well, #39 finally introduces a use case for sticking with `options`, and I plan to implement that before this.

So what should the API be for the arguments that come before the callback? Should it continue to just take an optional `options` hash? The problem w/ that is that most often the only "option" consumers want to pass is the `url`, so I'd rather have some way for consumers to do just that. I like how [esri-promise takes two optional arguments: a `url` string and a `dojoConfig` hash](https://github.com/nicksenger/esri-promise#new-in-02x), but I think I prefer to take a single optional argument that if is a string it is just a shortcut for `options.url`, but maybe that's too trixy.

Finally, I _think_ `promisify`/`denodify` will work w/ all those optional arguments, but I don't want to spend a lot of time testing that, so I may go straight to #8 instead of this once we decide on the API.

**Comment by tomwayson** [comment]
closing this in favor of #8 which should be resolved in #48
